### PR TITLE
modify session.execute() output and update user example

### PR DESCRIPTION
**Pull request by Aiee**
1. Change the output of session.execute() from ExecutionResponse to ResultSet
2. Update user example
3. Add String() for ValueWrapper and tests
3. Add MarshalJson() for ResultSet. To get the string value in the nested JSON struct, decode with base64

fix:
1. keep vertex/edge/path string format consistent with https://github.com/vesoft-inc/nebula-console/pull/52

### CODE REVIEW

File: go/value_wrapper.go [hunk 1]
```diff
@@ -238,3 +238,36 @@ func (valueWrapper ValueWrapper) GetType() string {
 	}
 	return "empty"
 }
+
+func (valWarp ValueWrapper) printValue() {
```
Comment (yixinglu) [inline go/value_wrapper.go]: suggest to provide a `ToString` function

File: go/logger.go [hunk 2]
```diff
@@ -20,17 +20,17 @@ type Logger interface {
 type DefaultLogger struct{}
 
 func (l DefaultLogger) Info(msg string) {
-	log.Printf("[INFO] %s", msg)
+	log.Printf("\n[INFO] %s", msg)
```
Comment (laura-ding) [inline go/logger.go]: log.Printf("[INFO] %s\n", msg)

File: go/logger.go [hunk 3]
```diff
@@ -20,17 +20,17 @@ type Logger interface {
...
 }
 
 func (l DefaultLogger) Warn(msg string) {
-	log.Printf("[WARNING] %s", msg)
+	log.Printf("\n[WARNING] %s", msg)
```
Comment (laura-ding) [inline go/logger.go]: ditto

File: go/logger.go [hunk 4]
```diff
@@ -20,17 +20,17 @@ type Logger interface {
...
 }
 
 func (l DefaultLogger) Error(msg string) {
-	log.Printf("[ERROR] %s", msg)
+	log.Printf("\n[ERROR] %s", msg)
```
Comment (laura-ding) [inline go/logger.go]: ditto

File: go/logger.go [hunk 5]
```diff
@@ -20,17 +20,17 @@ type Logger interface {
...
 }
 
 func (l DefaultLogger) Fatal(msg string) {
-	log.Fatalf("[FATAL] %s", msg)
+	log.Fatalf("\n[FATAL] %s", msg)
```
Comment (laura-ding) [inline go/logger.go]: ditto

File: go/result_set.go [hunk 6]
```diff
@@ -212,10 +211,6 @@ func (res ResultSet) GetRowValuesByIndex(index int) (*Record, error) {
 
 // Returns all rows
 func (res ResultSet) GetRows() []*nebula.Row {
-	if res.resp.Data == nil || res.resp.Data.Rows == nil {
```
Comment (laura-ding) [inline go/result_set.go]: why delete the check null data

File: go/result_set.go [hunk 7]
```diff
@@ -228,10 +223,6 @@ func (res ResultSet) GetErrorCode() graph.ErrorCode {
 }
 
 func (res ResultSet) GetErrorMsg() []byte {
-	if res.resp.ErrorMsg == nil {
```
Comment (laura-ding) [inline go/result_set.go]: ditto

File: go/value_wrapper.go [hunk 8]
```diff
@@ -238,3 +241,120 @@ func (valueWrapper ValueWrapper) GetType() string {
...
+	} else if value.IsSetFVal() {
+		fStr := strconv.FormatFloat(value.GetFVal(), 'f', -1, 64)
+		if !strings.Contains(fStr, ".") {
+			fStr = fStr + ".0"
+		}
+		return fStr
+	} else if value.IsSetSVal() {
+		return `"` + string(value.GetSVal()) + `"`
+	} else if value.IsSetDVal() {
+		return value.GetDVal().String()
```
Comment (laura-ding) [inline go/value_wrapper.go]: why the Date can directly to call it's String()

File: go/value_wrapper.go [hunk 9]
```diff
@@ -238,3 +241,120 @@ func (valueWrapper ValueWrapper) GetType() string {
...
+	} else if value.IsSetTVal() {
+		return value.GetTVal().String()
```
Comment (laura-ding) [inline go/value_wrapper.go]: ditto

File: go/value_wrapper.go [hunk 10]
```diff
@@ -238,3 +241,120 @@ func (valueWrapper ValueWrapper) GetType() string {
...
+	} else if value.IsSetDtVal() {
+		return value.GetDtVal().String()
```
Comment (laura-ding) [inline go/value_wrapper.go]: ditto

File: go/value_wrapper.go (hunk 8 above)
Comment (Aiee) [reply go/value_wrapper.go]: fixed. was thinking to add String() for time/date/dateTime when we have wrappers for these values

File: go/value_wrapper.go [hunk 11]
```diff
@@ -266,12 +266,17 @@ func (valWarp ValueWrapper) String() string {
...
-	} else if value.IsSetTVal() {
-		return value.GetTVal().String()
-	} else if value.IsSetDtVal() {
-		return value.GetDtVal().String()
+	} else if value.IsSetDVal() { // Date yyyy-mm-dd
+		date := value.GetDVal()
+		return fmt.Sprintf("%d-%d-%d", date.Year, date.Month, date.Day)
+	} else if value.IsSetTVal() { // Time HH:MM:SS:MS
+		time := value.GetTVal()
+		return fmt.Sprintf("%d:%d:%d:%d", time.Hour, time.Minute, time.Sec, time.Microsec)
```
Comment (laura-ding) [inline go/value_wrapper.go]: %d:%d:%d.%d

File: go/value_wrapper.go [hunk 12]
```diff
@@ -266,12 +266,17 @@ func (valWarp ValueWrapper) String() string {
...
+	} else if value.IsSetDtVal() { // DateTime yyyy-mm-dd HH:MM:SS:MS  TODO: add time zone
+		dateTime := value.GetDtVal()
+		return fmt.Sprintf("%d-%d-%d %d:%d:%d:%d",
```
Comment (laura-ding) [inline go/value_wrapper.go]: %d-%d-%dT%d:%d:%d.%d

**Review by yixinglu — DISMISSED** [review]

**Review by yixinglu — DISMISSED** [review]

File: go/result_set.go [hunk 13]
```diff
@@ -223,8 +232,22 @@ func (res ResultSet) GetColNames() []string {
 	return res.columnNames
 }
 
-func (res ResultSet) GetErrorCode() graph.ErrorCode {
-	return res.resp.ErrorCode
+// Returns an integer renprensting an error type
```
Comment (yixinglu) [inline go/result_set.go]: typo: representing

Comment (Aiee) [reply go/result_set.go]: fixed. Thanks for catching

File: go/value_wrapper.go [hunk 14]
```diff
@@ -16,6 +19,24 @@ type ValueWrapper struct {
 	value *nebula.Value
 }
 
+type ErrorCode int64
+
+const (
```
Comment (laura-ding) [inline go/value_wrapper.go]: why the errorcode in value_wrapper

**Review by laura-ding — APPROVED** [review]

**Review by jievince — APPROVED** [review]

**Pull request merged**